subsys/net/ip/dhcpv4.c

```diff
@@ -1071,7 +1077,9 @@ static enum net_verdict net_dhcpv4_input(struct net_conn *conn,
 
 	net_pkt_unref(pkt);
 
+	k_mutex_lock(&lock, K_FOREVER);
```

Comment:
The `dhcpv4_handle_reply()` already does locking so this should not be needed.

Reply:
Removed.

IMO it would be helpful if functions more clearly documented their interaction with the lock, which is why I've done so for the ones on the code paths relevant to `dhcpv4_timeout`.  If you'd rather not have those I'll remove them from the PR.